parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -291,32 +301,56 @@ private WriteBuilder createContextFunc(Function<Map<String, String>, Context> ne
       private final int dictionaryPageSize;
       private final CompressionCodecName codec;
       private final String compressionLevel;
+      private final int rowGroupCheckMinRecordCount;
+      private final int rowGroupCheckMaxRecordCount;
 
       private Context(int rowGroupSize, int pageSize, int dictionaryPageSize,
-                      CompressionCodecName codec, String compressionLevel) {
+                      CompressionCodecName codec, String compressionLevel,
+                      int rowGroupCheckMinRecordCount, int rowGroupCheckMaxRecordCount) {
         this.rowGroupSize = rowGroupSize;
         this.pageSize = pageSize;
         this.dictionaryPageSize = dictionaryPageSize;
         this.codec = codec;
         this.compressionLevel = compressionLevel;
+        this.rowGroupCheckMinRecordCount = rowGroupCheckMinRecordCount;
+        this.rowGroupCheckMaxRecordCount = rowGroupCheckMaxRecordCount;
       }
 
       static Context dataContext(Map<String, String> config) {
-        int rowGroupSize = Integer.parseInt(config.getOrDefault(
-            PARQUET_ROW_GROUP_SIZE_BYTES, PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT));
+        int rowGroupSize = PropertyUtil.propertyAsInt(config,
+            PARQUET_ROW_GROUP_SIZE_BYTES, PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT);
+        Preconditions.checkArgument(rowGroupSize > 0,
+            "Row group size must be > 0");
```

Comment:
Does this need to be on a separate line?

Reply:
fixed